_posts/2023-04-17-corewcf-1_4_0_preview.markdown

```diff
@@ -0,0 +1,145 @@
+
+Adding NetNamedPipe support to an app is similar to how you add NetTcp support. The api pattern has evolved a bit to make dealing with base paths a bit easier.  
+
+```csharp
+var builder = WebApplication.CreateBuilder();
+builder.WebHost.UseNetNamedPipe(options =>
+{
+    options.Listen("net.pipe://localhost/MyService.svc");
+});
+
+builder.Services.AddServiceModelServices();
+
+var app = builder.Build();
+app.UseServiceModel(serviceBuilder =>
+{
+    serviceBuilder.AddService<Service>();
+    serviceBuilder.AddServiceEndpoint<Service, IService>(new NetNamedPipeBinding(), "/netpipe");
+});
+app.Run();
+```
+
+There are overloads to the `Listen` method which takes another delegate to configure some settings which used to exist on NetNamedPipeBinding or NamedPipeTransportBindingElement but make more sense to be placed as listen options when configuring the WebHost. This api pattern matches the who you configure Kestrel. An application which uses that might look like this.
+
+```csharp
+var builder = WebApplication.CreateBuilder();
+builder.WebHost.UseNetNamedPipe(options =>
+{
+    options.Listen(new Uri("net.pipe://localhost/service.svc/"), listenOptions =>
+    {
+        // Set connection buffer size to 64K
+        listenOptions.ConnectionBufferSize = 64 * 1024;
+    });
+});
+```
+
+There are a couple of items needing to be completed before named pipe support will come out of preview. The biggest one is that there's currently no WSDL support. The second one is when stopping the WebApplication, the channel close doesn't happen cleanly. Instead of informing the client that the channel is closing, the underlying named pipe connection is abruptly closed.
+
+### Queue based transports
+
+When discussing bringing MSMQ support in a CoreWCF issue, I mentioned I'd like to make something generic so that CoreWCF could support multiple queue transport protocols. There was a lot of positive response to this idea, so I wrote up a brief description of how that could look. Biroj Nayak (@birojnayak) from Amazon AWS took this description and after a few discussions wrote a detailed design document. Dmitry Maslov (@Ximik87) submitted a PR with an initial implementation of this base queue support along with an MSMQ implementation built on top of it. Jon Louie (@jonlouie) from Amazon AWS implemented the RabbitMQ transport on top of this. Biroj helped iter with help from Biroj due to the initial base queue support needing some changes to accomodate different API patterns of the RabbitMQ client library. There are two other queue transports coming soon. The first will be for Apache Kafka and is being contributed by Guillaume Delahaye (@g7ed6e). This will be part of and released by the CoreWCF project. The second will be for Azure Queue Storage which is being developed by Microsoft and will be released by them.  
```

Comment:
```suggestion
When discussing bringing MSMQ support in a CoreWCF issue, I mentioned I'd like to make something generic so that CoreWCF could support multiple queue transport protocols. There was a lot of positive response to this idea, so I wrote up a brief [description](https://github.com/CoreWCF/CoreWCF/issues/164#issuecomment-908793395) of how that could look. Biroj Nayak (@birojnayak) from Amazon AWS took this description and after a few discussions wrote a detailed design [document](https://github.com/CoreWCF/CoreWCF/blob/main/Documentation/DesignDocs/corewcfqueue-design.md) . Dmitry Maslov (@Ximik87) submitted a PR with an initial implementation of this base queue support along with an MSMQ implementation built on top of it. Jon Louie (@jonlouie) from Amazon AWS implemented the RabbitMQ transport on top of this. Biroj helped iterate the base queue implementation due to needing some changes to accommodate different API patterns of the RabbitMQ client library. There are two other queue transports coming soon. The first will be for Apache Kafka and is being contributed by Guillaume Delahaye (@g7ed6e). This will be part of and released by the CoreWCF project. The second will be for Azure Queue Storage which is being developed by Microsoft and will be released by them.  
```